src/types/resolveStatements.ts

```diff
@@ -185,12 +188,16 @@ function processStatements(
                         s.ref,
                     );
                 }
-                sctx = addVariable(s.name, variableType, sctx);
+                if (s.name !== "_") {
+                    sctx = addVariable(s.name, variableType, sctx);
```

Comment:
we should probably move the check `s.name !== "_"` inside `addVariable` itself -- it would just ignore it